### ISSUE TITLE
jjMAP (polynomial map) is leaking memory

### ISSUE THREAD

**Issue by dimpase** [Singular/Singular#1090]
Running the following script 
```
ring r;
map F=r,x+y+z3,y+z+x2z3,z+1+xyz;
poly f=(x+y+z+xz)^10;
matrix m=f;
matrix mm;
while (1) {mm=F(m);}
```
doubles memory consumption every 30 sec or so. We suspect it's caused by a memory leak in `fast_map_common_subexp()` - and it was not fixed by the fix for #1089.

**Comment by dimpase** [comment]
just in case, I also mentioned this here: https://www.singular.uni-kl.de:8002/trac/ticket/875

**Comment by fingolfin** [comment]
@hannes14 is this still happening?

**Comment by dimpase** [comment]
It certainly is present in Singular 4.2.1.

**Comment by fingolfin** [comment]
Still happens with latest spielwiese, too. I don't see it doubling memory usage every 30 seconds, though; it just seems to grow linearly, maybe 20-40 MB per second or so (which is bad enough, of course)

**Comment by dimpase** [comment]
The leak might have gotten slower, this can be the case. IIRC it was worse before 4.2.1.

**Comment by hannes14** [comment]
work-around: https://github.com/Singular/Singular/commit/a3c2d148f63eb8d89dc8e26eb3d09063b2fa6590

**Comment by dimpase** [comment]
This appears to be fixed in Singular 4.3.2.

More precisely, tested with [Gentoo's sci-mathematics/singular-4.3.2_p1](https://packages.gentoo.org/packages/sci-mathematics/singular)
